### PR TITLE
[KED-893] - Comparison mode tooltip

### LINKED ISSUE #893: No tooltips on comparison mode icons
## Description

When comparing multiple runs side by side, the ‘baseline’ and ‘remove run’ icons do not show any tooltips. User has to guess what it is.

Screenshot 2022-05-24 at 13 54 1

Screenshot 2022-06-08 at 11 51 1

Screen Recording 2022-06-08 at 11 51 1

## Acceptance criteria

The circular ‘baseline’ and ‘remove run’ icons do not show any tooltips in the same style we use in all other icons across the product.

## Checklist
- [x] Include labels so that we can categorise your issue

### PR DESCRIPTION
**Pull request by Huongg**
## Description

- Fixes #893
- I also introduce a new prop for the `IconButton` called `labelTextPosition` to give us the flexibility to display the tooltip in a different direction (eg: top, bottom, right, left)

## Development notes

ezgif com-gif-maker (1)

## QA notes
**Tests:** 

- I've included tests to ensure the className reflects the value comes from `labelTextPosition`

**Design questions for @Mackay031 and @GabrielComymQB :**
- The tooltip currently displays at the bottom, a similar position to the bookmark, does it look okay to you two? Or you'd prefer to show it on the top? The only issue with showing on the top is sometimes it gets covered by the notification on the top right corner. 
- I can't find a specific design/ copy for the "Remove button" so I use "Remove run" as mentioned in the Github issue
- Also I include a tooltip for Baseline too, not sure if this is okay?

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/919"></a>

### CODE REVIEW

**Comment by GabrielComymQB** [comment]
Hey, I think it looks good. Here are my considerations.

Screenshot 2022-06-17 at 10 02 26

Screenshot 2022-06-17 at 10 05 36


When the "burger menu" is open and you try to hover over the other icons around, the baseline icon tooltip appears behind the dropdown menu. whereas the "remove run" icon appears at the front. Let's make sure they all appear in front of the dropdown.

**Comment by GabrielComymQB** [comment]
Another detail here I just noticed and this might be related to a different ticket is that, while the dropdown is active, the "burger menu" (three-dotted icon) should show its active state. At the moment the active state is only applied when you are hovering on it. 

Screenshot 2022-06-17 at 10 07 43

Basically if the dropdown is active everything is active including the icon.

Screenshot 2022-06-17 at 10 08 47

If this is not the right space for this change, can someone please point it in the right direction? Cheers

**Comment by cvetankanechevska** [comment]
> Another detail here I just noticed and this might be related to a different ticket is that, while the dropdown is active, the "burger menu" (three-dotted icon) should show its active state. At the moment the active state is only applied when you are hovering on it.
> 
> Screenshot 2022-06-17 at 10 07 43
> 
> Basically if the dropdown is active everything is active including the icon.
> 
> Screenshot 2022-06-17 at 10 08 47
> 
> If this is not the right space for this change, can someone please point it in the right direction? Cheers

@GabrielComymQB,  I've worked on this active state for the icons, and that was merged yesterday.

**Comment by GabrielComymQB** [comment]
Alright, thanks @cvetankanechevska !

**Review by GabrielComymQB — APPROVED** [review]

**Review by tynandebold — APPROVED** [review]
Nicely done! Just two tiny nits. Nothing blocking.

**Review by cvetankanechevska — APPROVED** [review]

File: src/components/ui/icon-button/icon-button.js [hunk 1]
```diff
@@ -3,6 +3,8 @@ import PropTypes from 'prop-types';
 import classnames from 'classnames';
 import './icon-button.css';
 
+const labelPositionTypes = ['right', 'left', 'bottom', 'top'];
```
[the remark this answers is not in the document]
Comment (Huongg) [reply src/components/ui/icon-button/icon-button.js:6]: thank you 😄

File: src/components/experiment-tracking/run-metadata/run-metadata.js [hunk 2]
```diff
@@ -151,15 +152,21 @@ const RunMetadata = ({
                           }
                           onClick={() => setPinnedRun(run.id)}
                           visible={enableShowChanges}
+                          labelText={
+                            run.id === pinnedRun ? 'Baseline' : 'Make baseline'
+                          }
+                          labelTextPosition="bottom"
```
[the remark this answers is not in the document]
Comment (Huongg) [reply src/components/experiment-tracking/run-metadata/run-metadata.js:158]: done 👍 thank you

**Pull request merged**